azext_edge/edge/providers/orchestration/resources/connector/opcua/certs.py

```diff
@@ -221,20 +217,27 @@ def client_add(
         resource_group: str,
         public_key_file: str,
         private_key_file: str,
-        subject_name: str,
-        application_uri: str,
         overwrite_secret: bool = False,
+        subject_name: Optional[str] = None,
+        application_uri: Optional[str] = None,
         public_key_secret_name: Optional[str] = None,
         private_key_secret_name: Optional[str] = None,
     ) -> dict:
         # inform user if the provided cert was issued by a CA, the CA cert must be added to the issuers list.
-        logger.warning("Please ensure the certificate must be added to the issuers list if it was issued by a CA.")
+        logger.warning(
+            "If this certificate was issued by a CA, then please ensure that the CA certificate is "
```

Comment:
Maybe in a later iteration there can be logic on when this warning is raised. Something like determining if the cert was self-signed or not, if not raise the warning.

Reply:
good idea, will add a TODO